### PR TITLE
liblsl: New port

### PR DESCRIPTION
**Pull request by adarshp**
###### Type(s)

- [ ] bugfix
- [X] enhancement
- [ ] security fix

###### Tested on

macOS 12.6.1 21G217 arm64
Xcode 14.2 14C18

###### Verification 
Have you

- [X] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [X] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [X] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [X] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [X] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`? [N/A, no existing tests]
- [X] tried a full install with `sudo port -vst install`?
- [X] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken? (N/A)

### CODE REVIEW

**Comment by herbygillot** [comment]
The build is [failing with](https://github.com/macports/macports-ports/actions/runs/4589736848/jobs/8104828928?pr=18168#step:8:334):

```
sh: /opt/local/bin/cmake: No such file or directory
```

I think what's happening is that you've overwritten `depends_build`.  The `cmake` portgroup adds `cmake` to the `depends_build` list, but your redefinition in your Portfile overwrites it with your own list.

I would recommend 2 changes:
1. Use the `boost` portgroup instead of explicitly specifying `boost` as a build dependency:
    ```
    PortSystem          1.0
    PortGroup           github 1.0
    PortGroup           cmake 1.1
    PortGroup           boost 1.0
    ```
    Once you specify this portgroup, remove `port:boost` from `depends_build`.
2. Use `depends_build-append` instead of `depends_build`:
    ```
    depends_build-append \
                   port:pugixml
    ```

File: science/liblsl/Portfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           cmake 1.1
+
+license             https://github.com/sccn/liblsl/blob/master/LICENSE
```
Comment (herbygillot) [inline science/liblsl/Portfile:7]: This is an **invalid** value for license.

The allowed values should be an identifier that identifies which kind of license this software has. You can see example values here: https://trac.macports.org/wiki/PortfileRecipes#licensekeyword

Taking a look, it looks like this should be the MIT license:

```suggestion
license             MIT
```

Comment (adarshp) [reply science/liblsl/Portfile]: Thanks! I've updated the license.

**Comment by adarshp** [comment]
Thanks for your feedback @herbygillot ! I've made the requested changes.

**Comment by herbygillot** [comment]
Thank you @adarshp, the build succeeded. Only nitpick would be to align the `port:pugixml` line with everything else, but LGTM otherwise.

**Comment by adarshp** [comment]
@herbygillot - no problem, I've pushed a commit to align that line.

**Comment by herbygillot** [comment]
Looks good and merged - thank you for your contribution, @adarshp.

**Pull request merged**